src/Microsoft.AspNetCore.Razor.Language/Extensions/IMetadataIdentifierFeature.cs

```diff
@@ -0,0 +1,10 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNetCore.Razor.Language.Extensions
+{
+    internal interface IMetadataIdentifierFeature : IRazorEngineFeature
+    {
+        string GetIdentifier(RazorCodeDocument codeDocument, RazorSourceDocument sourceDocument);
```

Comment:
Is `RazorCodeDocument` even necessary in this api?

Reply:
It's not necessary in the default implementation

Reply:
Oops I meant `RazorSourceDocument`. Why not use `codeDocument.Source`?

Reply:
This applies to imports too

Reply:
Ah. Got it 👍